qiling/os/windows/dlls/kernel32/processenv.py

```diff
@@ -37,6 +37,25 @@ def hook_GetCommandLineW(ql, address, params):
     return addr
 
 
+
+
+# LPWSTR * CommandLineToArgvW(
+#    LPCWSTR lpCmdLine,
+#    int     *pNumArgs
+# );
+
+# Note: Didn't test it. Need second thoughts.
+@winsdkapi(cc=STDCALL, dllname=dllname)
+def hook_CommandLineToArgvW(ql, address, params):
+        cmdline = ql.loader.cmdline + params["lpCmdLine"]
+        cmdline = ql.loader.cmdline.decode('ascii').encode('utf-16le')
+        numArgs = ql.os.heap.alloc(len(cmdline))
+        ptrs = ql.mem.write(numArgs, cmdline)
+        return ptrs
```

Comment:
This implementation is incorrect.
It should parse the passed in string in 'lpCmdLine'.
Only if that is an empty string will it return the current process executable as only argument.

And it parses the `lpCmdline` to split it into multiple arguments, which is not something you do here.

Reply:
See https://git.reactos.org/?p=reactos.git;a=blob;f=dll/win32/shell32/wine/shell32_main.c;hb=9c8ed68544f1adf49fb941d83a807159433e4323#l79 for an example c implementation